aten/src/ATen/native/quantized/cpu/qconcat.cpp

```diff
@@ -0,0 +1,69 @@
+#include <ATen/ATen.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+
+#include <algorithm>
+#include <vector>
+
+namespace at { namespace native {
+namespace {
+
+bool is_valid_quantization_type(Tensor t) {
+  const auto qtype = t.qscheme();
+  return (qtype == kPerTensorAffine) ||
+         (qtype == kPerTensorSymmetric);
+}
+
+template <bool ReLUFused = false>
+class QCat final : public torch::OperatorKernel {
+ public:
+  Tensor operator()(const std::vector<Tensor>& qxs, int64_t axis,
+                    c10::optional<double> scale,
+                    c10::optional<int64_t> zero_point) {
+    TORCH_CHECK(is_valid_quantization_type(qxs[0]),
+                "Only per-tensor quantization is supported in 'cat'!")
+    const auto x_dtype = qxs[0].scalar_type();
+    const auto x_scale = qxs[0].q_scale();
+    const auto x_zero_point = qxs[0].q_zero_point();
+    double _scale = scale ? *scale : qxs[0].q_scale();
+    int64_t _zero_point = zero_point ? *zero_point : qxs[0].q_zero_point();
+
+    std::vector<Tensor> xs;
+    xs.reserve(qxs.size());
+    for (const auto& qx: qxs) {
+      TORCH_CHECK(is_valid_quantization_type(qx),
+                  "Only per-tensor quantization is supported in 'cat'!")
+      TORCH_CHECK(x_dtype == qx.scalar_type(), "All dtypes must be the same.");
+      TORCH_CHECK(x_scale == qx.q_scale(), "All scales must be the same.");
```

Comment:
do we only want to support concatenating tensors with same scale and zero_point? is that to strict?

Reply:
Oh, I misread one of your older comments.